connectors/sources/slack.py

```diff
@@ -0,0 +1,382 @@
+        This is hard, because no one property is reliably present and optimal.
+        This function goes through a variety of options, from most normalized and readable to least.
+        :param user:
+        :return: an identifier for the user - hopefully a human-readable name.
+        """
+        user_profile = user.get("profile", {})
+        if user_profile.get("display_name_normalized"):
+            return user_profile["display_name_normalized"]
+        elif user_profile.get("real_name_normalized"):
+            return user_profile["real_name_normalized"]
+        elif user.get("real_name"):
+            return user["real_name"]
+        elif user.get("name"):
+            return user["name"]
+        else:
+            return user[
+                "id"
+            ]  # Some Users do not have any names (like Bots). For these, we fall back on ID
+
+    def remap_user(self, user):
+        user_profile = user.get("profile", {})
+        return {
+            "_id": user["id"],
+            "type": "user",
+            "real_name_normalized": user_profile.get("real_name_normalized"),
+            "profile_real_name": user_profile.get("real_name"),
+            "display_name": user_profile.get("display_name"),
+            "display_name_normalized": user_profile.get("display_name_normalized"),
+            "first_name": user_profile.get("first_name"),
+            "last_name": user_profile.get("last_name"),
+        } | dict_slice(
+            user,
+            (
+                "id",
+                "name",
+                "deleted",
+                "real_name",
+                "is_admin",
+                "is_owner",
+                "is_bot",
+            ),
+        )
+
+    def remap_channel(self, channel):
+        topic = channel.get("topic", {})
+        topic_creator_id = topic.get("creator")
+        purpose = channel.get("purpose", {})
+        purpose_creator_id = purpose.get("creator")
+        return {
+            "_id": channel["id"],
+            "type": "channel",
+            "topic": topic.get("value"),
+            "topic_author": self.usernames.get(topic_creator_id, topic_creator_id),
+            "topic_last_updated": topic.get("last_set"),
+            "purpose": purpose.get("value"),
+            "purpose_author": self.usernames.get(
+                purpose_creator_id, purpose_creator_id
+            ),
+            "purpose_last_updated": purpose.get("last_set"),
+            "last_updated": channel.get("updated"),
+        } | dict_slice(
+            channel,
+            (
+                "id",
+                "name",
+                "name_normalized",
+                "created",
+            ),
+        )
+
+    def remap_message(self, message, channel):
+        user_id = message.get("user", message.get("bot_id"))
+
+        def convert_usernames(
+            match_obj,
+        ):  # used below in an re.sub to map on matching messages
+            id = match_obj.group(1)
```

Comment:
I think we should not assign to built-in variable `id`

Reply:
Good call, renamed